### PR TITLE
Notebook for SPARCL + Jdaviz

### PR DESCRIPTION
**Pull request by weaverba137**
This PR adds a how-to for displaying SPARCL spectra in Jdaviz (specifically Specivz).  This notebook was initially developed at [.Astronomy 12](https://www.dotastronomy.com/twelve/).

### CODE REVIEW

**Comment by jacquesalice** [comment]
Hi @weaverba137 I've gone through and added section descriptions and other small formatting edits. Jdaviz is available on gp02 if you would like to review/test it on there. @rnikutta and @galaxyumi , can yall take a look at this notebook as well?

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+     "shell.execute_reply.started": "2023-10-10T19:54:13.337382Z"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# SPARCL import\n",
+    "from sparcl.client import SparclClient\n",
+    "# Jdaviz import\n",
+    "from jdaviz import Specviz\n",
```
Comment (galaxyumi) [inline 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:125]: It seems that the module 'jdaviz' has not been installed by default for everyone. I am encountering the following error when running this cell: 'No module named 'jdaviz'.' Could you either make the package installation global or add instructions on how to install this module for individual accounts at the beginning of this NB?

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8f2b74af-e9ea-4845-bb6d-38ae1425067a",
+   "metadata": {},
+   "source": [
+    "<a class=\"anchor\" id=\"searchsparcl\"></a>\n",
+    "## Search for spectra in SPARCL\n",
+    "Using the `client.find()` method, we obtain the following fields from the DESI EDR data set using SPARCL:  \n",
+    "- `sparcl_id` : Universally Unique Identifier for spectrum in SPARCL\n",
```
Comment (galaxyumi) [inline 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:219]: lower case 'u'?

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+    "- `ra` : Right Ascension in degrees\n",
```
Comment (galaxyumi) [inline 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:220]: lower case 'a'?

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c3123870-6933-43a5-b79f-184b8f58ff41",
+   "metadata": {},
+   "source": [
+    "<a class=\"anchor\" id=\"retrievesparcl\"></a>\n",
+    "## Retrieve spectra from SPARCL and prepare plot data\n",
+    "Here we define a function that:\n",
+    "1. Uses the `client.retrieve()` method from SPARCL to retrieve spectra data for the list of IDs from DESI EDR.\n",
```
Comment (galaxyumi) [inline 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:322]: delete 'data'

**Review by galaxyumi — COMMENTED** [review]
@weaverba137 and @jacquesalice Because the module 'jdaviz' was not installed for my account, my current review is very limited to text. I should be able to provide more complete feedback once you figure out a solution regarding the 'jdaviz' installation.

**Comment by galaxyumi** [comment]
@weaverba137 and @jacquesalice, here are more comments:

1. The last two commente-out lines in the cell where you define the function `on_selected` can be deleted if they are no longer in use.
2. The last cell for making plot could have more description in the markdown. For example, featuring a drop down bar where one can select a target to display its spectrum with the best-fit model. 
3. The y-axis label is not following the scientific unit style conventions, e.g.,[erg s^{-1} cm^{-2} A^{-1}]. 
4. The x-axis ticks are too busy to read. Maybe changing to nm? I am getting a long warning box complaining about the Angstrom anyway. 
5. It would be great if the plot's legend could be more informative. Currently, they are random(?) two alphabets in red and black colors for model and data.  
6. Please take care of this long warning if possible. Otherwise, please leave a comment about the warnings in the markdown not to scare the users. Screen Shot 2023-10-16 at 4 50 06 PM

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "934ba639-42a1-4649-a64b-25f8d33c9ad0",
+   "metadata": {},
+   "source": [
+    "<a class=\"anchor\" id=\"intro\"></a>\n",
+    "## Introduction\n",
+    "\n",
+    "This notebook demonstrates how to find and retrieve spectroscopy data for certain objects from the DESI EDR data set using [SPARCL](https://astrosparcl.datalab.noirlab.eduhttps://astrosparcl.datalab.noirlab.edu) (SPectra Analysis and Retrievable Catalog Lab) and producing an interactive plot of an object's spectrum using [Jdaviz](https://jdaviz.readthedocs.io/en/latest/index.html).\n",
```
[the remark this answers is not in the document]
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:67]: Fixed.

[the remark this answers is not in the document]
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:67]: Fixed.

[the remark this answers is not in the document]
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:67]: "display"

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb (hunk 1 above)
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:125]: It should now be installed on both `gp02` and `gp12`.

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb (hunk 2 above)
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:219]: "Universally Unique Identifier" is a genuine proper noun.

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb (hunk 3 above)
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:220]: "Right Ascension" is valid. Among other things, it's abbreviated `RA`, not `Ra`.

File: 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb (hunk 4 above)
Comment (weaverba137) [reply 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:322]: Fixed.

**Comment by weaverba137** [comment]
From the comments above:

1. Fixed.
2. Fixed.
3. Waiting on a fully viable notebook server to test. This may be an internal feature of Jdaviz that cannot be changed.
4. Waiting on a fully viable notebook server to test. This may be an internal feature of Jdaviz that cannot be changed.
5. The alphabetical indexes are an internal feature of Jdaviz. We cannot change that in this notebook.
6. Waiting on a fully viable notebook server to fix.

**Comment by weaverba137** [comment]
Updates:

* The warnings are now fixed.
* The y-axis label looks fine to me.
* The x-axis looks fine to me.

@jacquesalice, I think this is ready to go, but it should be tested on `gp02` when it becomes available again, and also tested on `gp12` when Jdaviz is installed on that system.

**Comment by weaverba137** [comment]
PS, I updated the HTML version on the last commit. Please double-check that it looks OK.

**Comment by weaverba137** [comment]
I've retested this notebook with the new 'Python 3' kernel on `gp02`. Please review again. There are no major changes from last time, but it has been a while since we were able to test this.

**Comment by stephjuneau** [comment]
Hi @weaverba137 , @jacquesalice , @galaxyumi : checking on the status of this notebook. From Ben's last comment it was retested on gp02. What are the next steps after reviewing this version before it can be deployed to prod/gp12?

**Comment by weaverba137** [comment]
I believe it requires the Python 3.10 kernel to be installed on the production server, so that might also be a question for @mjfitzpatrick.

**Comment by weaverba137** [comment]
@jacquesalice, it should be possible to at least review this notebook on gp13 so that it is ready to merge if not yet fully merged.

**Review by jacquesalice — APPROVED** [review]
Notebook runs well on gp13. Approving this PR and awaiting merge until we switch over to gp13

**Pull request merged**